open-build-service/notqmail.spec.in

```diff
@@ -0,0 +1,365 @@
+%verify (not user group mode caps) %attr(4755, qmailq, qmail)   %{qmaildir}/bin/qmail-queue
+%else
+%attr(4755,qmailq,qmail)           %{qmaildir}/bin/qmail-queue
+%endif
+
+%files doc
+%defattr(-, root, root,-)
+%dir %attr(0755,root,root) %{nq_docdir}
+     %attr(0644,root,root) %{nq_docdir}/*
+%attr(0644,root,root) %{_mandir}/man?/*%{?ext_man}
+
+### SCRIPTLETS ###############################################################################
+%verifyscript
+%if 0%{?suse_version} >= 1120
+  %verify_permissions -e %{qmaildir}/bin/qmail-queue
+  %verify_permissions -e %{qmaildir}/alias/
+  %verify_permissions -e %{qmaildir}/queue/lock/trigger
+%endif
+/var/qmail/bin/instcheck
+
+%pretrans
+argv1=$1
+if test -f %{_sysconfdir}/systemd/system/multi-user.target.wants/%{name}.service
+then
+  echo "Giving %{name} exactly 5 seconds to exit nicely"
+  /bin/systemctl stop %{name} > /dev/null 2>&1
+fi
+sleep 5
+
+%if 0%{?fedora_version} || 0%{?centos_version} || 0%{?rhel_version} || 0%{?suse_version} < 1500 || 0%{?sles_version} < 15
+%pre
+  /usr/bin/getent group nofiles   > /dev/null || /usr/sbin/groupadd nofiles
+  /usr/bin/getent group qmail     > /dev/null || /usr/sbin/groupadd qmail
+  /usr/bin/getent passwd alias    > /dev/null || /usr/sbin/useradd -M -g nofiles  -d %{qmaildir}/alias  -s /sbin/nologin alias
+  /usr/bin/getent passwd qmaild   > /dev/null || /usr/sbin/useradd -M -g nofiles  -d %{qmaildir}        -s /sbin/nologin qmaild
+  /usr/bin/getent passwd qmaill   > /dev/null || /usr/sbin/useradd -M -g nofiles  -d %{logdir}          -s /sbin/nologin qmaill
+  /usr/bin/getent passwd qmailp   > /dev/null || /usr/sbin/useradd -M -g nofiles  -d %{qmaildir}        -s /sbin/nologin qmailp
+  /usr/bin/getent passwd qmailq   > /dev/null || /usr/sbin/useradd -M -g qmail    -d %{qmaildir}        -s /sbin/nologin qmailq
+  /usr/bin/getent passwd qmailr   > /dev/null || /usr/sbin/useradd -M -g qmail    -d %{qmaildir}        -s /sbin/nologin qmailr
+  /usr/bin/getent passwd qmails   > /dev/null || /usr/sbin/useradd -M -g qmail    -d %{qmaildir}        -s /sbin/nologin qmails
+for i in alias qmaild qmaill qmailp qmailq qmailr qmails
+do
+  %{__rm} -f /var/spool/mail/$i
+done
+%else
+%pre -f notqmail.pre
+%endif
+%if 0%{?suse_version} || 0%{?sles_version}
+  %service_add_pre %{name}.service
+%endif
+
+### SCRIPTLET ###############################################################################
+%post
+%if 0%{?suse_version} >= 1120
+%if 0%{?set_permissions:1} > 0
+  %set_permissions %{qmaildir}/bin/qmail-queue
+  %set_permissions %{qmaildir}/alias/
+%else
+  %run_permissions
+%endif
+%endif
+/var/qmail/bin/instpackage queue-only
+/var/qmail/bin/instchown queue-only
+%if 0%{?suse_version} || 0%{?sles_version}
+  %service_add_post %{name}.service
+%endif
+
+if [ ! -f %{qmaildir}/control/me ] ; then
+  uname -n > %{qmaildir}/control/me
+fi
+
+# setup notqmail as alternative mta
+alternatives --install /usr/sbin/sendmail mta %{qmaildir}/bin/sendmail 120 \
```

Comment:
Not a blocker, but for maintenance, will it be possible later to extract common subroutines shared across packages so they can be called uniformly (and without duplication) from .e.g `.deb` and `.rpm` packaging?

Reply:
This is very much required. I have this in mind. If it is made into a common script, there will be few things that will need to be done depending on whether it is a RPM or a debian system. e.g. updating notqmail as an alternative is different for RPM and debian based system. If we use initscripts in /etc/init.d, /etc/rc*.d, then the alternatives command becomes a bit different on openSUSE systems. So I will have to have few if else conditions in the script. e.g. `if [ -d /etc/debian ] ; then ...`

But, yes making this common makes it less prone to mistakes when changes are made

RPM
```
alternatives --install /usr/sbin/sendmail mta /var/qmail/bin/sendmail 120 \
  --slave /usr/lib/sendmail mta-sendmail /var/qmail/bin/sendmail
```

Debian
```
update-alternatives --install \
  /usr/sbin/sendmail mta /var/qmail/bin/sendmail \
  120 --slave /usr/lib/sendmail mta-sendmail \
  /var/qmail/bin/sendmail
```